optimum/habana/transformers/generation/utils.py

```diff
@@ -2982,16 +2982,20 @@ def _sample(
             eos_positions_tmp = torch.isin(
                 input_ids[:, start_token_idx:], torch.tensor(eos_token_id).to(device=input_ids.device)
             ).int()
-            if eos_positions_tmp.numel() != 0:
-                # argmax(dim=1) is throwing this error in eager mode, if the tensor is empty
-                eos_positions = eos_positions + eos_positions_tmp.argmax(dim=1)
-
-            # Create a mask for positions greater than the first eos_token_id
-            mask = torch.arange(generation_config.max_length, device="hpu").expand(
-                batch_size, generation_config.max_length
-            ) > eos_positions.unsqueeze(1)
-            # Apply the mask to set positions greater than the first eos_token_id to pad_token_id
-            input_ids[mask] = pad_token_id
+            if torch.all(eos_positions_tmp == 0):
```

Comment:
This will return True if `eos_positions_tmp ` is empty it seems, not sure that's the behaviour we want. Related to the comment at line 2991. @12010486 @yafshar What do you think about this change? Since this is related to the PR we merged last week

Reply:
From my understanding, if eos_positions_tmp is an all-zero tensor, it means the generated tokens doesn't contain eos_token, i.e. the generation is stopped by max length instead of eos token, so there is no need to do the masking by eos_token.